### PR TITLE
Update to Parquet 1.12.0

### PR DESCRIPTION
**Pull request by electrum**
Depends on https://github.com/trinodb/trino-hive-apache/pull/28

### CODE REVIEW

**Comment by electrum** [comment]
@raunaqmorarka I released `hive-apache` with your change

**Comment by raunaqmorarka** [comment]
> @raunaqmorarka I released `hive-apache` with your change

@electrum Thanks !

File: lib/trino-parquet/src/main/java/io/trino/parquet/ParquetTypeUtils.java [hunk 1]
```diff
@@ -149,6 +149,9 @@ public static ParquetEncoding getParquetEncoding(Encoding encoding)
                 return ParquetEncoding.PLAIN;
             case RLE:
                 return ParquetEncoding.RLE;
+            case BYTE_STREAM_SPLIT:
+                // TODO: implement support for this encoding
```
Comment (hashhar) [inline lib/trino-parquet/src/main/java/io/trino/parquet/ParquetTypeUtils.java:153]: Let's create an issue and link it here so we don't forget.

**Review by hashhar — APPROVED** [review]
LGTM % comments.

**Pull request merged**